virttest/qemu_devices/qdevices.py

```diff
@@ -1504,6 +1573,36 @@ def verify_hotplug(self, out, monitor):
                 return True
         return False
 
+    def _cmdline_json(self):
+        command_dict = {}
+        out = "-%s " % self.type
+        params = self.params.copy()
+        params = self._convert_memobj_args(params)
+        command_dict["qom-type"] = params["backend"]
+        del params["backend"]
+        return out + '\'' + json.dumps(dict(command_dict, **params)) + '\''
+
+    @staticmethod
+    def _convert_memobj_args(args):
+        """
+        Convert string type of string to uint64, and create new dict
+        (e.g: "size": "14336M" ) from old dict(e.g: "size": 15032385536)
+        to qmp syntax.
+
+        :param args: Dictionary with the qmp parameters.
+        :type args: dict
+        :return: Converted args.
+        :rtype: dict
+        """
+        new_args = dict()
+        for key, value in args.items():
+            if key == "size":
+                new_args[key] = int(utils_numeric.normalize_data_size(value,
+                                                                      "B"))
+            else:
+                new_args[key] = value
+        return new_args
```

Comment:
if 'size' in args:
    args['size'] = int(utils_numeric.normalize_data_size(args['size'], B)) ?

So we don't need a loop copy of other items?

Reply:
Sorry that I lost this part in my previous review, anyway, yours is still OK for me.